src/lib/parallel_scan/parallel_scan.ml

```diff
@@ -55,154 +49,128 @@ end
  * Each node has a weight associated to it and the 
  * new jobs received are distributed across the tree based on this number. *)
 module Weight = struct
+  [%%versioned
   module Stable = struct
     module V1 = struct
-      module T = struct
-        type t = {base: int; merge: int}
-        [@@deriving sexp, bin_io, version {unnumbered}]
-      end
+      type t = {base: int; merge: int} [@@deriving sexp]
 
-      include T
+      let to_latest = Fn.id
     end
+  end]
 
-    module Latest = V1
-  end
+  type t = Stable.Latest.t = {base: int; merge: int} [@@deriving sexp]
 
-  type t = Stable.Latest.t = {base: int; merge: int} [@@deriving sexp, lens]
+  (* code that would have been generated by `deriving lens`
+     workaround for interaction between %%versioned and lens.ppx_deriving
+     see: https://discuss.ocaml.org/t/odd-deriving-error/4606
+   *)
```

Comment:
Can you add a TODO here, so that we know to come back and fix it when ocaml-ppx/ppx_deriving#203 is fixed?

Reply:
Done.